//------------------------------------------------------------------------------
// <auto-generated>
//     This code was generated by AsyncGenerator.
//
//     Changes to this file may cause incorrect behavior and will be lost if
//     the code is regenerated.
// </auto-generated>
//------------------------------------------------------------------------------


using System;
using NUnit.Framework;

namespace NHibernate.Test.NHSpecificTest.NH704
{
	using System.Threading.Tasks;
	using System.Threading;
	[TestFixture]
	public class FixtureAsync : BugTestCase
	{
		public override string BugNumber
		{
			get { return "NH704"; }
		}

		[Test]
		public async Task ReAttachCatTestAsync()
		{
			try
			{
				using (ISession sess = OpenSession())
				{
					Cat c = new Cat();
					await (sess.SaveAsync(c, CancellationToken.None));
					await (sess.FlushAsync(CancellationToken.None));
					sess.Clear();
					await (sess.LockAsync(c, LockMode.None, CancellationToken.None)); //Exception throw here
				}
			}
			finally
			{
				using (ISession s = OpenSession())
				{
					await (s.DeleteAsync("from Cat", CancellationToken.None));
					await (s.FlushAsync(CancellationToken.None));
				}
			}
		}
	}
}